app/src/main/java/org/gateshipone/odyssey/viewitems/ListViewItem.java

```diff
@@ -253,6 +253,10 @@ public void setAlbumTrack(final Context context, final TrackModel track, final b
         // duration
         String trackDuration = FormatHelper.formatTracktimeFromMS(context, track.getTrackDuration());
 
+        if (trackTitle.isEmpty()) {
+            trackTitle = track.getTrackTitle();
```

Comment:
It would be a better solution to just modify line 240 to 242 to 
trackTitle += track.getTrackTitle()
and remove the local trackName variable.

Reply:
Hi,
Thanks for your reply.

If I'm not mistaken, your solution means that if the file has a disc number and/or a track number but not a track name (yes, I agree, this would be weird), this would lead to a display like:

    discNumber.trackNumber - filename.ogg

Another solution might be the following:

Before line 228 (``if (showDiscNumber)``):

```java
if (trackName.isEmpty()) {
    trackTitle = track.getTrackTitle();
} else {
```

and  ``}`` instead of lines 256-258 (and of course removing the ``if (!trackName.isEmpty()) {`` test in between)

Reading this patch again, I realize that, as is, there are two things that are called title. This is actually a bit confusing.

Should the part of this function that computes ``trackTitle`` belong to the TrackModel class? One solution would be to make ``getTrackTitle`` actually return trackTitle as computed in ``setAlbumTrack`` but this will change what is displayed in the notifications. Which might be desirable, the same thing gets displayed everywhere (but the notification suddenly becomes more verbose). If so, should ``trackTitle`` be computed in the constructor of ``TrackModel``?

What do you think?

Reply:
> If I'm not mistaken, your solution means that if the file has a disc number and/or a track number but not a track name (yes, I agree, this would be weird), this would lead to a display like:

> discNumber.trackNumber - filename.ogg

 I think this would be correct and should be done like this. I don't see a problem there.

> Should the part of this function that computes trackTitle belong to the TrackModel class? One solution would be to make getTrackTitle actually return trackTitle as computed in setAlbumTrack but this will change what is displayed in the notifications. Which might be desirable, the same thing gets displayed everywhere (but the notification suddenly becomes more verbose). If so, should trackTitle be computed in the constructor of TrackModel?  

I agree that trackTitle vs trackName is confusing. I don't like the idea of more information in the notification/widget as they are very constrained in terms of visible space and the track (disc) number is of no use there. Perhaps getTrackTitle could be renamed to ensure the difference to trackName is more clear?

Reply:
All right, what about ``getTrackDisplayedName`` then, instead of ``getTrackTitle``?
I'll make a commit to change this with your approval.

Reply:
Sounds good

Reply:
I pushed this.
Thanks to both of you for your time in handling this pull request!